pages/author/[slug].js

```diff
@@ -1,125 +1,134 @@
-                        <div className="text-gray-500 flex">
-                          <span>{post.publishedAt}</span>
-                          <span className=" after:content-['\00B7'] after:mx-1 "></span>
-                          <span>{post.readingTime}</span>
+                  <div className="flex flex-col justify-between flex-[1_0_0%] px-2.5 py-5 sm:p-5 rounded-b-lg">
+                    <a
+                      href={"/post/" + post.slug}
+                      className="text-black text-xl"
+                    >
+                      {post.title}
+                    </a>
+                    <div>
+                      <MarkdownView
+                        className="text-gray-800 mt-5 text-sm line-clamp-3"
+                        markdown={post.content}
+                      />
+                      <div className="pt-4">
+                        <div className="flex flex-row ">
+                          <div className="text-gray-500 flex">
+                            <span>{post.publishedAt}</span>
+                            <span className=" after:content-['\00B7'] after:mx-1 "></span>
+                            <span>{post.readingTime}</span>
+                          </div>
                         </div>
                       </div>
                     </div>
                   </div>
                 </div>
-              </div>
-            );
-          })}
-        </div>
+              );
+            })}
+          </div>
+        )}
       </div>
     </section>
   );
 };
 
 export async function getStaticPaths() {
   const Authors = await fetchAuthor();
-  const paths = Authors.data.map((Author) => ({
-    params: { slug: Author.attributes.slug },
-  }));
-  return {
-    paths: paths,
-    fallback: false,
-  };
+  if (Authors.data) {
+    const paths = Authors.data.map((Author) => ({
+      params: { slug: Author.attributes.slug },
+    }));
+    return {
+      paths: paths,
+      fallback: false,
+    };
+  }
 }
 
 export async function getStaticProps(context) {
   const slug = context.params.slug;
-
   if (!slug) {
     throw new Error("Slug not valid");
   }
 
   var author = await fetchAuthor(slug);
-  author = author.data.attributes;
-
-  author["totalStory"] = author.posts.data.length;
-  for (let i = 0; i < author.posts.data.length; i++) {
-    const post = author.posts.data[i].attributes;
-    post.publishedAt = await formateDate(post.publishedAt);
-    post["readingTime"] = await getReadingTime(post.content);
-  }
+  if (author.data) {
+    author = author.data.attributes;
 
-  if (!author) {
-    return { notFound: true };
+    author["totalStory"] = author.posts.data.length;
+    for (let i = 0; i < author.posts.data.length; i++) {
+      const post = author.posts.data[i].attributes;
+      post.publishedAt = await formateDate(post.publishedAt);
+      post["readingTime"] = await getReadingTime(post.content);
+    }
+  } else {
+    console.log(author.error.status);
```

Comment:
If data is not there, show the message on UI(Can be popup or simple message) like `There are no authors` or `authors not found`